### PR TITLE
Alg mongodb integration

### LINKED ISSUE #4: Implement MongoDB
Set up mongoDB to handle articles for PlanetSide. 

Model should follow standard for any blog. Google it

### PR DESCRIPTION
**Pull request by gaaliciA1990**
Implement MongoDB #4 

Completed the KMongoDB integration into the application. Add a data package with a data class. Updated all of the Controller classes to work with the DB for all CRUD actions. Refactored some code for easier testing and readablility

### CODE REVIEW

File: src/main/kotlin/com/gamenews/Application.kt [hunk 1]
```diff
@@ -12,6 +14,8 @@ fun main() {
 }
 
 fun Application.module() {
-    configureRouting()
+    val controller: Controller = Controller(ArticlesDatabase())
```
Comment (CRamsan) [inline src/main/kotlin/com/gamenews/Application.kt]: Let's split this into two separate lines.

File: src/main/kotlin/com/gamenews/data/ArticlesDatabase.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import com.gamenews.models.Article
+import org.litote.kmongo.coroutine.coroutine
+import org.litote.kmongo.reactivestreams.KMongo
+
+/**
+ * Class for our database queries
+ */
+class ArticlesDatabase {
+    private val client = KMongo.createClient().coroutine
+    private val database = client.getDatabase("ArticlesDatabase")
```
Comment (CRamsan) [inline src/main/kotlin/com/gamenews/data/ArticlesDatabase.kt]: This database should also be passed as an argument to the `ArticleDatabase`. That will making it easier to test.

Comment (CRamsan) [reply src/main/kotlin/com/gamenews/data/ArticlesDatabase.kt]: Something like:
```
class ArticlesDatabase(
    private val database: Database<Article>,
) {...} 
```

File: src/main/kotlin/com/gamenews/Application.kt (hunk 1 above)
Comment (CRamsan) [reply src/main/kotlin/com/gamenews/Application.kt]: For example:
```
val articlesDatabase = ArticlesDatabase()
val controller = Controller(articlesDatabase)
```

File: src/main/kotlin/com/gamenews/models/Article.kt [hunk 3]
```diff
@@ -1,31 +1,24 @@
...
     companion object {
         // automatically generate unique ID
-        private val idCounter = AtomicInteger()
+        private val idCounter = ObjectId().toString()
 
         /**
          * Create a new article with an id, title, and body
          */
-        fun newEntry(title: String, body: String) = Article(idCounter.getAndIncrement(), title, body)
+        fun newEntry(title: String, body: String) = Article(idCounter, title, body)
```
Comment (CRamsan) [inline src/main/kotlin/com/gamenews/models/Article.kt]: I have the suspicion that `idCounter` will have the same value for every entity (insert thinking face). Could we verify that the `idCounter` changes for every new entry that is created.

File: src/main/kotlin/com/gamenews/plugins/Controller.kt [hunk 4]
```diff
@@ -14,24 +15,39 @@ import io.ktor.util.pipeline.PipelineContext
...
             )
         )
     }
 
     /**
      * Handles calls for creating a new article
      */
-    suspend fun newArticle(context: PipelineContext<Unit, ApplicationCall>) {
+    suspend fun createNewArticle(context: PipelineContext<Unit, ApplicationCall>) {
         context.call.respond(
```
Comment (CRamsan) [inline src/main/kotlin/com/gamenews/plugins/Controller.kt:50]: We need to call the database here to create the new article.

Comment (CRamsan) [reply src/main/kotlin/com/gamenews/plugins/Controller.kt:50]: Or I think that `saveArticle` is used for creating a new article? (thinking face). This displays the page to create a new article right?

File: src/main/kotlin/com/gamenews/Application.kt (hunk 1 above)
Comment (gaaliciA1990) [reply src/main/kotlin/com/gamenews/Application.kt]: Done

File: src/main/kotlin/com/gamenews/data/ArticlesDatabase.kt (hunk 2 above)
Comment (gaaliciA1990) [reply src/main/kotlin/com/gamenews/data/ArticlesDatabase.kt]: Done

File: src/main/kotlin/com/gamenews/models/Article.kt (hunk 3 above)
Comment (gaaliciA1990) [reply src/main/kotlin/com/gamenews/models/Article.kt]: Done

File: src/main/kotlin/com/gamenews/plugins/Controller.kt (hunk 4 above)
Comment (gaaliciA1990) [reply src/main/kotlin/com/gamenews/plugins/Controller.kt:50]: Done

**Review by CRamsan — APPROVED** [review]

**Pull request merged**